bundles/binding/org.openhab.binding.sonance/src/main/java/org/openhab/binding/sonance/internal/SonanceBinding.java

```diff
@@ -0,0 +1,447 @@
+							sendVolumeCommand(itemName, "FF550210" + group,
+									outputStreamCache.get(key),
+									bufferedReaderCache.get(key));
+					} catch (UnknownHostException e) {
+						logger.error("UnknownHostException occured.");
+					} catch (IOException e) {
+						logger.debug("Amplifier is offline, status can't be updated at this moment.");
+						try {
+							socketCache.get(key).close();
+						} catch (Exception ex) {
+						}
+						socketCache.remove(key);
+						outputStreamCache.remove(key);
+						bufferedReaderCache.remove(key);
+						offlineEndPoints.add(key); // Stop trying to fetch other
+													// values from this end
+													// point until next execute
+													// cycle
+					}
+				}
+			}
+		}
+	}
+
+	@Override
+	public void bindingChanged(BindingProvider provider, String itemName) {
+		super.bindingChanged(provider, itemName);
+
+		for (Map.Entry<String, Socket> entry : socketCache.entrySet())
+			try {
+				entry.getValue().close();
+			} catch (IOException e) {
+				logger.error("Can't close a socket when binding changed.");
+			}
+
+		// Cleanup all sockets
+		socketCache.clear();
+		outputStreamCache.clear();
+		bufferedReaderCache.clear();
+	}
+
+	@Override
+	protected void internalReceiveCommand(String itemName, Command command) {
+		logger.info("Command received ({}, {})", itemName, command);
+
+		SonanceBindingProvider provider = findFirstMatchingBindingProvider(itemName);
+		String group = provider.getGroup(itemName);
+		String ip = provider.getIP(itemName);
+		int port = provider.getPort(itemName);
+
+		Socket s = null;
+		try {
+			s = new Socket(ip, port);
+			DataOutputStream outToServer = new DataOutputStream(
+					s.getOutputStream());
+			BufferedReader i = new BufferedReader(new InputStreamReader(
+					s.getInputStream()));
+
+			if (provider.isMute(itemName))
+				switch (command.toString()) {
+				case "OFF":
+					sendMuteCommand(itemName, "FF550207" + group, outToServer,
+							i); // Mute on
+					break;
+				case "ON":
+					sendMuteCommand(itemName, "FF550208" + group, outToServer,
+							i); // Mute off
+					break;
+				default:
+					logger.error("I don't know what to do with this command");
+					break;
+				}
+			else if (provider.isVolume(itemName))
+				if (command.toString().equals("UP"))
+					sendVolumeCommand(itemName, "FF550204" + group,
+							outToServer, i);
+				else if (command.toString().equals("DOWN"))
+					sendVolumeCommand(itemName, "FF550205" + group,
+							outToServer, i);
+				else {
+					try {
+						setVolumeCommand(itemName, group,
+								Integer.parseInt(command.toString()),
+								outToServer, i, ip + ":" + port);
+					} catch (NumberFormatException nfe) {
+						logger.error("I don't know what to do with this command");
+					}
+
+				}
+			s.close();
+		} catch (IOException e) {
+			logger.debug("IO Exception when sending command");
+		} finally {
+			try {
+				if (s != null)
+					s.close();
+			} catch (IOException e) {
+			}
```

Comment:
HI Thomas

 

Do you mean I need to add this extra class (http://ejoe.sourceforge.net/035/apidocs/de/netseeker/ejoe/io/IOUtils.html) just for having 2 lines less? Could I get an example in other OpenHab code somewhere?

 

Kind regards

Laurens Van Acker 

 

From: Thomas Eichstädt-Engelen [mailto:notifications@github.com] 
Sent: zaterdag 26 december 2015 21:24
To: openhab/openhab <openhab@noreply.github.com>
Cc: Laurens <laurens@van.acker.be>
Subject: Re: [openhab] Sonance binding (#3401)

 

In bundles/binding/org.openhab.binding.sonance/src/main/java/org/openhab/binding/sonance/internal/SonanceBinding.java <https://github.com/openhab/openhab/pull/3401#discussion_r48449592> :

> +                                                          Integer.parseInt(command.toString()),
> +                                                          outToServer, i, ip + ":" + port);
> +                                   } catch (NumberFormatException nfe) {
> +                                           logger.error("I don't know what to do with this command");
> +                                   }
> +
> +                            }
> +                    s.close();
> +            } catch (IOException e) {
> +                    logger.debug("IO Exception when sending command");
> +            } finally {
> +                    try {
> +                            if (s != null)
> +                                   s.close();
> +                    } catch (IOException e) {
> +                    }

you could also use IOUtils.closeQuite()

—
Reply to this email directly or view it on GitHub <https://github.com/openhab/openhab/pull/3401/files#r48449592> .  <https://github.com/notifications/beacon/AELo6AlY6_GtDd_t0M1ZUp-KmrvMT50Pks5pTu7NgaJpZM4GefUp.gif>